---
title: '8\. Worksheet: Phylogenetic Diversity - Traits'
author: "Anna Lennon"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: pdf_document
geometry: margin=2.54cm
---

## OVERVIEW

Up to this point, we have been focusing on patterns taxonomic diversity in Quantitative Biodiversity. 
Although taxonomic diversity is an important dimension of biodiversity, it is often necessary to consider the evolutionary history or relatedness of species. 
The goal of this exercise is to introduce basic concepts of phylogenetic diversity. 

After completing this exercise you will be able to:

1.  create phylogenetic trees to view evolutionary relationships from sequence data
2.  map functional traits onto phylogenetic trees to visualize the distribution of traits with respect to evolutionary history
3.  test for phylogenetic signal within trait distributions and trait-based patterns of biodiversity

## Directions:
1. In the Markdown version of this document in your cloned repo, change "Student Name" on line 3 (above) with your name.
2. Complete as much of the worksheet as possible during class.
3. Use the handout as a guide; it contains a more complete description of data sets along with examples of proper scripting needed to carry out the exercises.
4. Answer questions in the  worksheet.
Space for your answers is provided in this document and is indicated by the ">" character.
If you need a second paragraph be sure to start the first line with ">".
You should notice that the answer is highlighted in green by RStudio (color may vary if you changed the editor theme). 
5. Before you leave the classroom, **push** this file to your GitHub repo.
6. For the assignment portion of the worksheet, follow the directions at the bottom of this file. 
7. When you are done, **Knit** the text and code into a PDF file.
8. After Knitting, submit the completed exercise by creating a **pull request** via GitHub.
Your pull request should include this file `PhyloTraits_Worskheet.Rmd` and the PDF output of `Knitr` (`PhyloTraits_Worskheet.pdf`).

The completed exercise is due on **Wednesday, February 26^th^, 2025 before 12:00 PM (noon)**.

## 1) SETUP

In the R code chunk below, provide the code to:  
1. clear your R environment,  
2. print your current working directory,  
3. set your working directory to your `Week6-PhyloTraits/` folder, and  
4. load all of the required R packages (be sure to install if needed).  

```{r}
rm(list=ls())
getwd()
setwd("/cloud/project/QB2025_ALennon/Week6-PhyloTraits")
# package.list <- c('ape', 'seqinr', 'phylobase', 'adephylo', 'geiger', 'picante', 'stats', 'RColorBrewer', 'caper', 'phylolm', 'pmc', 'ggplot2', 'tidyr', 'dplyr', 'phangorn', 'pander') 
# for (package in package.list) {
#   if (!require(package, character.only=TRUE, quietly=TRUE)) {
#     install.packages(package)
#     library(package, character.only=TRUE)
#   }
# }
# install.packages("pak")
# #comment out after first run to prevent continuous reinstall/update
# pak::pkg_install("msa")
# 
# if (!require("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("Biostrings")
library(bios2mds)
library(Biostrings)
library(ape)
library(seqinr)
library(phylobase)
library(adephylo)
library(geiger)
library(picante)
library(stats)
library(RColorBrewer)
library(phylolm)
library(caper)
library(pmc)
library(ggplot2)
library(dplyr)
library(tidyr)
library(phangorn)
library(pander)
library(msa)

```

## 2) DESCRIPTION OF DATA

The maintenance of biodiversity is thought to be influenced by **trade-offs** among species in certain functional traits. 
One such trade-off involves the ability of a highly specialized species to perform exceptionally well on a particular resource compared to the performance of a generalist. 
In this exercise, we will take a phylogenetic approach to mapping phosphorus resource use onto a phylogenetic tree while testing for specialist-generalist trade-offs. 

## 3) SEQUENCE ALIGNMENT

***Question 1***: Using your favorite text editor, compare the `p.isolates.fasta` file and the `p.isolates.afa` file. 
Describe the differences that you observe between the two files. 

> ***Answer 1***: Both the fasta and afa file contain a series of nucleotides (ACGT) color coded. However, the fasta file does not contain any gaps and the output is much more extensive compared the afa file that is cmpressed to view the last line of nucleotides (1500) for the samples and contains the gaps present in the genome.

In the R code chunk below, do the following:
1. read your alignment file,
2. convert the alignment to a DNAbin object,
3. select a region of the gene to visualize (try various regions), and
4. plot the alignment using a grid to visualize rows of sequences.

```{r}
seq.fasta <-readDNAStringSet("data/p.isolates.fasta", format = "fasta")
seq.afa <-readDNAStringSet("data/p.isolates.afa", format = "fasta")

# seq.fasta
# seq.afa

read.aln.fasta <-msaMuscle(seq.fasta)
save.aln.fasta <-msaConvert(read.aln.fasta, type = "bios2mds::align")
export.fasta(save.aln.fasta, "./data/p.isolates.2.afa")
p.DNAbin.fasta <-as.DNAbin(read.aln.fasta)
window.L <-p.DNAbin.fasta[,100:1000]
image.DNAbin(window.L, cex.lab = .5)
```

***Question 2***:  Make some observations about the `muscle` alignment of the 16S rRNA gene sequences for our bacterial isolates and the outgroup, *Methanosarcina*, a member of the domain Archaea. 
Move along the alignment by changing the values in the `window` object. 

a. Approximately how long are our sequence reads?  
b. What regions do you think would are appropriate for phylogenetic inference and why?  

> ***Answer 2a***:By changing the window amount, the length of the sequence reads are around 800 base pairs. However, some of the sequences are smaller (~700 bp).
> ***Answer 2b***: Regions with similar base pairs reveals conserved regions where as high amounts of color diversity could insiuate regions under selective pressures. At about 460 bp, there is a region of highly conserved genes with some variation (green) that could be interesting to make phylogenetic inferences. Comparatively, at about 580 bp there is a region of alternating blue-green that could insight a region under selective pressures that would be fascinating to study. 

## 4) MAKING A PHYLOGENETIC TREE

Once you have aligned your sequences, the next step is to construct a phylogenetic tree.
Not only is a phylogenetic tree effective for visualizing the evolutionary relationship among taxa, but as you will see later, the information that goes into a phylogenetic tree is needed for downstream analysis. 

### A. Neighbor Joining Trees

In the R code chunk below, do the following:  
1. calculate the distance matrix using `model = "raw"`,  
2. create a Neighbor Joining tree based on these distances,  
3. define "Methanosarcina" as the outgroup and root the tree, and  
4. plot the rooted tree. 

```{r}
seq.dist.raw <- dist.dna(p.DNAbin.fasta, model = "raw", pairwise.deletion = FALSE)
#tree
nj.tree <- bionj(seq.dist.raw)
outgroup <- match("Methanosarcina", nj.tree$tip.label)
nj.rooted <- root(nj.tree, outgroup, resolve.root = TRUE)


par(mar = c(1, 1, 2, 1) + 0.1)
plot.phylo(nj.rooted, main = "Neighbor Joining Tree of Isolates", "phylogram", 
           use.edge.length = FALSE, direction = "right", cex = 0.6, 
           label.offset = 1)
add.scale.bar(cex = 0.7)

```

***Question 3***: What are the advantages and disadvantages of making a neighbor joining tree?   


> ***Answer 3***: Neighbor-joining trees are the simplest form of phyolgenetic tree and very computationally light. These tree allow one to view broad patterns. However, the raw estimates of phylogeentic distance are used, meaning they assume that substitutions only occur once and at the same rate. This is not accurate to biology as base pairs can be substituted multiple different times and at different rates. The resulting tree may not be accurate to the actual life history as a result. 
 


### B) SUBSTITUTION MODELS OF DNA EVOLUTION 

In the R code chunk below, do the following:  
1. make a second distance matrix based on the Felsenstein 84 substitution model,  
2. create a saturation plot to compare the *raw* and *Felsenstein (F84)* substitution models,  
3. make Neighbor Joining trees for both, and  
4. create a cophylogenetic plot to compare the topologies of the trees.

```{r}
# F84 matrix
seq.dist.F84 <- dist.dna(p.DNAbin.fasta, model = "F84", pairwise.deletion = FALSE)
#matrix comparison
par(mar = c(5, 5, 2, 1) + 0.1)
plot(seq.dist.raw, seq.dist.F84, 
     pch = 20, col = "red", las = 1, asp = 1, xlim = c(0, 0.7), 
     ylim = c(0, 0.7), xlab = "Raw Distance", ylab = "F84 Distance") 
abline(b = 1, a = 0, lty = 2)
text(0.65, 0.6, "1:1")

#trees
raw.tree <- bionj(seq.dist.raw)
F84.tree <- bionj(seq.dist.F84)

raw.outgroup <- match("Methanosarcina", raw.tree$tip.label) 
F84.outgroup <- match("Methanosarcina", F84.tree$tip.label) 

raw.rooted <- root(raw.tree, raw.outgroup, resolve.root = TRUE) 
F84.rooted <- root(F84.tree, F84.outgroup, resolve.root = TRUE)

layout(matrix(c(1, 2), 1, 2), width = c(1, 1))
par(mar = c(1, 1, 2, 0))
plot.phylo(raw.rooted, type = "phylogram", direction = "right", 
      show.tip.label = TRUE, use.edge.length = FALSE, adj = 0.5, 
      cex = 0.6, label.offset = 2, main = "Raw")
par(mar = c(1, 0, 2, 1))
plot.phylo(F84.rooted, type = "phylogram", direction = "left", 
      show.tip.label = TRUE, use.edge.length = FALSE, adj = 0.5, 
      cex = 0.6, label.offset = 2, main = "F84")
#tree length diff
dist.topo(raw.rooted, F84.rooted, method = "score")
```

### C) ANALYZING A MAXIMUM LIKELIHOOD TREE

In the R code chunk below, do the following:  
1. Read in the maximum likelihood phylogenetic tree used in the handout.
2. Plot bootstrap support values onto the tree

```{r}
phyDat.aln <- msaConvert(read.aln.fasta, type = "phangorn::phyDat")

# NJ-ML method
aln.dist <- dist.ml(phyDat.aln)
aln.NJ <- NJ(aln.dist)

fit <- pml(tree = aln.NJ, data = phyDat.aln)

# JC69 model
fitJC <- optim.pml(fit, TRUE)

# GTR model
fitGTR <- optim.pml(fit, model = "GTR", optInv = TRUE, optGamma = TRUE, 
          rearrangement = "NNI", control = pml.control(trace = 0))

# Perform model selection with either an ANOVA test or with AIC
anova(fitJC, fitGTR)
AIC(fitJC)
AIC(fitGTR)
#Bootstrap
ml.bootstrap <- read.tree("./data/ml_tree/RAxML_bipartitions.T1")
par(mar = c(1, 1, 2, 1) + 0.1)
plot.phylo(ml.bootstrap, type = "phylogram", direction = "right", 
      show.tip.label = TRUE, use.edge.length = FALSE, cex = 0.6, 
      label.offset = 1, main = "Maximum Likelihood with Support Values")
add.scale.bar(cex = 0.7)
nodelabels(ml.bootstrap$node.label, font = 2, bg = "white", 
      frame = "r", cex = 0.5)


```

***Question 4***:  

a) How does the maximum likelihood tree compare the to the neighbor-joining tree in the handout? 
If the plots seem to be inconsistent with one another, explain what gives rise to the differences.

b) Why do we bootstrap our tree?

c) What do the bootstrap values tell you? 

d) Which branches have very low support? 

e) Should we trust these branches? Why or why not?

> ***Answer 4a***: The maximum likelihood tree and the NJ tree vary from one another. Given that the NJ tree does not account for nucleotide states and not based on statistics compared to Neighborhood joining tree, trees produced by ML are typically more robust. 
> ***Answer 4b***: Maximum likelihood trees are based on statistical computation that finds the tree with the highest probability of being the true phylogeny. Bootstrapping assigns values of support to indicate how reliable/resolve a branch is with hgher values indicating more resolved branches. 
> ***Answer 4c***: High bootstrap values indicate more reliable/resolved branches. Therefore, values with 95% or higher are considered resolved compared to values beteern 94-70 having some support and less than 50 are considered unresolved. 
> ***Answer 4d***: There are several key brnaches with low support. In particular, there is a cluster of branches at the top of the tree analyzing WG42-LL1 relationship, the node between WG74-WG21, WG58B-WG27.  
> ***Answer 4e***: The low values indicate that there is low resolution for these branches. Therefore, these branches should be handeled with caution. Overall, the tree has relatively high bootstrap values, so the tree as a whole is relatively resolved. However, these low resolution branches should not be taken as operationally correct and assumed as unresolved.  


## 5) INTEGRATING TRAITS AND PHYLOGENY

### A. Loading Trait Database

In the R code chunk below, do the following:  
1. import the raw phosphorus growth data, and  
2. standardize the data for each strain by the sum of growth rates.

```{r}
p.growth <- read.table("./data/p.isolates.raw.growth.txt", sep = "\t", 
            header = TRUE, row.names = 1)

# Standardize 
p.growth.std <- p.growth / (apply(p.growth, 1, sum))

```

### B. Trait Manipulations

In the R code chunk below, do the following:  
1. calculate the maximum growth rate ($\mu_{max}$) of each isolate across all phosphorus types,  
2. create a function that calculates niche breadth (*nb*), and  
3. use this function to calculate *nb* for each isolate.

```{r}
umax <- (apply(p.growth, 1, max)) 

#function for niche breadth (nb)
nb.levins <- function(p_xi = ""){
  p = 0
  for (i in p_xi){
    p = p + i^2
    }
  nb = 1 / (length(p_xi) * p)
  return(nb)
}
#Isolate NB
nb <- as.matrix(nb.levins(p.growth.std))
nb <- setNames(as.vector(nb), as.matrix(row.names(p.growth)))

```  

### C. Visualizing Traits on Trees

In the R code chunk below, do the following:  
1. pick your favorite substitution model and make a Neighbor Joining tree,  
2. define your outgroup and root the tree, and  
3. remove the outgroup branch.

```{r}
seq.dist.T92 <-dist.dna(p.DNAbin.fasta, model = "T92", pairwise.deletion = FALSE)
nj.tree.T92 <-bionj(seq.dist.T92)
outgroup.T92 <- match("Methanosarcina", nj.tree$tip.label)
nj.rooted.T92 <- root(nj.tree.T92, outgroup.T92, resolve.root = TRUE)
nj.rooted.T92 <- drop.tip(nj.rooted.T92, "Methanosarcina")


#plot(nj.rooted)


```

In the R code chunk below, do the following:  
1. define a color palette (use something other than "YlOrRd"),  
2. map the phosphorus traits onto your phylogeny,  
3. map the *nb* trait on to your phylogeny, and  
4. customize the plots as desired (use `help(table.phylo4d)` to learn about the options).


```{r}
mypalette <- colorRampPalette(brewer.pal(9, "PRGn"))

nj.plot.T92 <- nj.rooted.T92
nj.plot.T92$edge.length <- nj.plot.T92$edge.length + 10^-1

par(mar = c(1, 1, 1, 1) + 0.1)
x <- phylo4d(nj.plot.T92, p.growth.std)
table.phylo4d(x, treetype = "phylo", symbol = "colors", show.node = TRUE, 
              cex.label = 0.5, scale = FALSE, use.edge.length = FALSE, 
              edge.color = "black", edge.width = 2, box = FALSE, 
              col=mypalette(25), pch = 15, cex.symbol = 1.25, 
              ratio.tree = 0.5, cex.legend = 1.5, center = FALSE)   

# Niche Breadth
par(mar = c(1, 5, 1, 5) + 0.1)
x.nb <- phylo4d(nj.plot.T92, nb)
table.phylo4d(x.nb, treetype = "phylo", symbol = "colors", show.node = TRUE, 
      cex.label = 0.5, scale = FALSE, use.edge.length = FALSE, 
      edge.color = "black", edge.width = 2, box = FALSE, col = mypalette(25), 
      pch = 15, cex.symbol = 1.25, var.label = ("NB"),ratio.tree = 0.90, 
      cex.legend = 1.5, center = FALSE)


```

***Question 5***:  

a) Develop a hypothesis that would support a generalist-specialist trade-off.

b) What kind of patterns would you expect to see from growth rate and niche breadth values that would support this hypothesis?

> ***Answer 5a***: Generalist species that occupy a wide variety traits will be better fit to a highly variable environments and be less fit in stable environments whereas specialist species will be better fit in highly stable environments and signifcantly less fit in highly variable environments. Therefore, species that occupy similar niches (generalist or specialist) will be more closely related.
> ***Answer 5b***: When testing ths hypothesis, individuals that are closely related should have similar niche breadth values. However, the growth rate for individuals with similar niche breadth values may be constrainted as species compete for similar resources. This may cause certain species to diverage into more specialist roles despite having sister taxa that are generalist. 

## 6) HYPOTHESIS TESTING

### Phylogenetic Signal: Pagel's Lambda 

In the R code chunk below, do the following:  
1. create two rescaled phylogenetic trees using lambda values of 0.5 and 0,   
2. plot your original tree and the two scaled trees, and  
3. label and customize the trees as desired.

```{r}
#original tree: nj.rooted.T92
nj.lambda.5 <- geiger:::rescale.phylo(nj.rooted, model = "lambda", lambda = 0.5)
nj.lambda.0 <-geiger:::rescale.phylo(nj.rooted, model = "lambda", lambda = 0)
layout(matrix(c(1, 2, 3), 1, 3), width = c(1, 1, 1))
par(mar=c(1, 0.5, 2, 0.5) + 0.1) 
plot(nj.rooted.T92, main = "lambda = 1", cex = 0.7, adj = 0.5)
plot(nj.lambda.5, main = "lamba = 0.5", cex = 0.7, adj = 0.5)
plot(nj.lambda.0, main = "lamba = 0",  cex = 0.7, adj = 0.5)

```

In the R code chunk below, do the following:  
1. use the `fitContinuous()` function to compare your original tree to the transformed trees.

```{r}
fitContinuous(nj.rooted.T92, nb, model = "lambda")
fitContinuous(nj.lambda.0, nb, model = "lambda")
phylosig(nj.rooted.T92, nb, method = "lambda", test = TRUE)

#Phylo signal
nj.rooted.T92$edge.length <- nj.rooted.T92$edge.length + 10^-7

p.phylosignal <- matrix(NA, 6, 18)
colnames(p.phylosignal) <- colnames(p.growth.std)
rownames(p.phylosignal) <- c("K", "PIC.var.obs", "PIC.var.mean", 
                             "PIC.var.P", "PIC.var.z", "PIC.P.BH")

for (i in 1:18){
  x <- setNames(as.vector(p.growth.std[,i]),row.names(p.growth))
  out <- phylosignal(x, nj.rooted.T92)
  p.phylosignal[1:5, i] <- round(t(out), 6)
}

p.phylosignal[6, ] <- round(p.adjust(p.phylosignal[4, ], method = "BH"), 3)

print(p.phylosignal)
```

***Question 6***:  There are two important outputs from the `fitContinuous()` function that can help you interpret the phylogenetic signal in trait data sets. 
a. Compare the lambda values of the untransformed tree to the transformed (lambda = 0).
b. Compare the Akaike information criterion (AIC) scores of the two models. 
Which model would you choose based off of AIC score (remember the criteria that the difference in AIC values has to be at least 2)?
c. Does this result suggest that there's phylogenetic signal?

> ***Answer 6a***: The untransformed tree lambda is .006947 compared the the transformed lambda which is .00179646.   
> ***Answer 6b***: The AIC value for the untransformed tree is -36.9997 while the AIC of the log tranformed model is -36.6647. Alithough the untranformed tree has a lower AIC value, the two models are considered equivalent because they do not have a difference of 2. 
> ***Answer 6c***: These results do not suggest there is a phylogenetic signal. 



## 7) PHYLOGENETIC REGRESSION

***Question 7***: 
In the R code chunk below, do the following:  
1. Clean the resource use dataset to perform a linear regression to test for differences in maximum growth rate by niche breadth and lake environment.
2. Fit a linear model to the trait dataset, examining the relationship between maximum growth rate by niche breadth and lake environment,
2. Fit a phylogenetic regression to the trait dataset, taking into account the bacterial phylogeny

```{r}
nb.lake = as.data.frame(as.matrix(nb))
nb.lake$lake = rep('A')

for(i in 1:nrow(nb.lake)) {
    ifelse(grepl("WG",row.names(nb.lake)[i]), nb.lake[i,2] <- "WG", 
        nb.lake[i,2] <- "LL")
}

colnames(nb.lake)[1] <- "NB"

#Max growth rate
umax <- as.matrix((apply(p.growth, 1, max))) 
nb.lake = cbind(nb.lake,umax)

# Plot 
ggplot(data = nb.lake,aes(x = NB, y = log10(umax), color = lake)) +
  geom_point() +
  geom_smooth(method = "lm") +
  xlab("Niche breadth") +
  ylab(expression(Log[10]~"(Maximum growth rate)"))

#Linear
fit.lm <- lm(log10(umax) ~ NB*lake,data = nb.lake)
summary(fit.lm)
AIC(fit.lm)

#Phylo corrected
fit.plm <- phylolm(log10(umax) ~ NB*lake, data = nb.lake, nj.rooted, 
          model = "lambda", boot = 0)
summary(fit.plm)
AIC(fit.plm)


```
  

a. Why do we need to correct for shared evolutionary history?
b. How does a phylogenetic regression differ from a standard linear regression?
c. Interpret the slope and fit of each model. 
Did accounting for shared evolutionary history improve or worsen the fit?
d. Try to come up with a scenario where the relationship between two variables would completely disappear when the underlying phylogeny is accounted for.

> ***Answer 7a***:Correcting for the shared evolutionary history is essential because the linear regression model assumes independence. However, in this data, the shared history means the variables are not independent so it needs to be corrected.   
> ***Answer 7b***: Phylogenetic regression is the varience of residual errors by the covariance matrix, which allows branch length to be factored into the model. Phylogenetic regression also takes into account phylogenetic signal when determining the covariance matrix. Simple linear regression only analyzes the relationship between residual error by a vairable.  
> ***Answer 7c***: For LL lake, the slope is near zero, therefore there is only a small portion of maximum growth explained by niche breadth. Comparatively, WG lake has a steep negative slope resulting in a negative relationship between niche breadth and maximum growth. Correcting for phylogeny made the fit worse (from 48.41 to 41.73)
> ***Answer 7d***: The relationship between two variables would completely disappear when doing a phylogenetic regression if one of the variables was highly correlated with phylogeny. For example, if analyzing the relationship between wing pattern of butterflies and niche breadth, wing patterns may be too similar among taxa that would result in no relationship if phylogeny was taken into account. 


## 7) SYNTHESIS

Work with members of your Team Project to obtain reference sequences for 10 or more taxa in your study.
Sequences for plants, animals, and microbes can found in a number of public repositories, but perhaps the most commonly visited site is the National Center for Biotechnology Information (NCBI) <https://www.ncbi.nlm.nih.gov/>.
In almost all cases, researchers must deposit their sequences in places like NCBI before a paper is published. 
Those sequences are checked by NCBI employees for aspects of quality and given an **accession number**.
For example, here an accession number for a fungal isolate that our lab has worked with: JQ797657.
You can use the NCBI program nucleotide **BLAST** to find out more about information associated with the isolate, in addition to getting its DNA sequence: <https://blast.ncbi.nlm.nih.gov/>.
Alternatively, you can use the `read.GenBank()` function in the `ape` package to connect to NCBI and directly get the sequence. 
This is pretty cool.
Give it a try. 

But before your team proceeds, you need to give some thought to which gene you want to focus on. 
For microorganisms like the bacteria we worked with above, many people use the ribosomal gene (i.e., 16S rRNA). 
This has many desirable features, including it is relatively long, highly conserved, and identifies taxa with reasonable resolution. 
In eukaryotes, ribosomal genes (i.e., 18S) are good for distinguishing course taxonomic resolution (i.e. class level), but it is not so good at resolving genera or species. 
Therefore, you may need to find another gene to work with, which might include protein-coding gene like cytochrome oxidase (COI) which is on mitochondria and is commonly used in molecular systematics. 
In plants, the ribulose-bisphosphate carboxylase gene (*rbcL*), which on the chloroplast, is commonly used.
Also, non-protein-encoding sequences like those found in **Internal Transcribed Spacer (ITS)** regions between the small and large subunits of of the ribosomal RNA are good for molecular phylogenies.
With your team members, do some research and identify a good candidate gene. 

After you identify an appropriate  gene, download sequences and create a properly formatted fasta file.
Next, align the sequences and confirm that you have a good alignment. 
Choose a substitution model and make a tree of your choice. 
Based on the decisions above and the output, does your tree jibe with what is known about the evolutionary history of your organisms? 
If not, why? 
Is there anything you could do differently that would improve your tree, especially with regard to future analyses done by your team?

```{r}
library(devtools)
devtools::install_github("jinyizju/V.PhyloMaker")
require(V.PhyloMaker)


setwd("/cloud/project/QB_biodiversity_project_EH/")
SPCDinfo <- read.csv("/cloud/project/QB_biodiversity_project_EH/SPCDinfo.csv")

treesp <- SPCDinfo[c("Scientific.Name", "Genus", "Family")] #extract relevant cols 
treephylo <- phylo.maker(treesp, scenarios = "S3", nodes = nodes.info.1) #make tree
plot.phylo(treephylo$scenario.3, cex = .4)
treephylo$scenario.3 

```

```{r}
setwd("/cloud/project/QB_biodiversity_project_EH/")
tree.fasta <-readDNAStringSet("/cloud/project/QB_biodiversity_project_EH/Tree_fasta.txt", format = "fasta")
read.aln.tree <-msaMuscle(tree.fasta)
save.aln.tree <-msaConvert(read.aln.tree, type = "bios2mds::align")

tree.DNAbin.fasta <-as.DNAbin(read.aln.tree)
window.L <-tree.DNAbin.fasta[,100:1000]
image.DNAbin(window.L, cex.lab = .5)

phyData.aln.tree <-msaConvert(read.aln.tree, type = "phangorn::phyDat")
aln.dist.tree <-dist.ml(phyData.aln.tree)
aln.NJ.tree <-NJ(aln.dist.tree)
fit.tree <- pml(tree = aln.NJ.tree, data = phyData.aln.tree)

# JC69 model
fitJC.tree <- optim.pml(fit.tree, TRUE)

# GTR model
fitGTR.tree <- optim.pml(fit.tree, model = "GTR", optInv = TRUE, optGamma = TRUE, 
          rearrangement = "NNI", control = pml.control(trace = 0))

# Perform model selection with either an ANOVA test or with AIC
anova(fitJC.tree, fitGTR.tree)
AIC(fitJC.tree)
AIC(fitGTR.tree)

#phylo tree: aln.NJ.tree


par(mar = c(1, 1, 2, 1) + 0.1)
plot.phylo(aln.NJ.tree, type = "phylogram", direction = "right", 
      show.tip.label = TRUE, use.edge.length = FALSE, cex = 0.6, 
      label.offset = 1, main = "Maximum Likelihood with Support Values")
add.scale.bar(cex = 0.7)
nodelabels(ml.bootstrap$node.label, font = 2, bg = "white", 
      frame = "r", cex = 0.5)

```
>tree matches known phylogeny. I do need to fix the labels.

## SUBMITTING YOUR ASSIGNMENT
Use Knitr to create a PDF of your completed `8.PhyloTraits_Worksheet.Rmd` document, push it to GitHub, and create a pull request.
Please make sure your updated repo include both the pdf and RMarkdown files.
Unless otherwise noted, this assignment is due on **Wednesday, February 26^th^, 2025 at 12:00 PM (noon)**.



